src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedRootModuleTypeStaticConstructor.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRootModuleTypeStaticConstructor : MethodSymbol
```

Comment:
>SynthesizedRootModuleTypeStaticConstructor [](start = 26, length = 42)

Why do we even need a symbol for this method? Do we intend to expose it outside of the compilation pipeline?

Reply:
I'm not sure if we intend to expose it. @RikkiGibson How would I evaluate that question?

@AlekseyTs Please see https://github.com/dotnet/roslyn/pull/43281#discussion_r407010880. The answer to "why" is that I don't have a way to find out what to do besides asking for reviews of this PR. I took the approach of copying what you did with the PrivateImplementationDetails class which synthesizes its static constructor the same way.

Reply:
@RikkiGibson I'm not sure of the ramifications of exposing symbols versus not. (Besides that testing could get harder.) Do you have an opinion or know what line of reasoning to apply?

Reply:
I like the idea that we don't need symbols until proven otherwise, so I'll go with that for now.